app/src/main/java/com/rikin/cinco/domain/GameViewModel.kt

```diff
@@ -184,26 +185,47 @@ class GameViewModel(private val clipboardHelper: ClipboardHelper) : ViewModel()
     }
 
     private fun updateKeyboard(row: RowState, keyboard: KeyboardState): KeyboardState {
-        val updatedRows = mutableListOf<KeyboardRowState>()
-        keyboard.keyRows.forEach { keyRow ->
-            val updatedKeys = mutableListOf<KeyState>()
-            keyRow.keys.forEach { key ->
-                // There is an issue here where a guess with a repeating letter in which the position
-                // of the first is incorrect, but the position of the second isn't can appear as
-                // incorrect. This is because we only look for the first letter in the row instead of
-                // all instances of a letter.
-                val tileForKey = row.tiles.find { it.letter == key.letter }
-                if (tileForKey != null && key.status.priority >= tileForKey.status.priority) {
-                    updatedKeys.add(key.copy(status = tileForKey.status))
-                } else {
-                    updatedKeys.add(key)
+
+        fun updateKeyMappings(): KeyboardState {
```

Comment:
With my suggestion, this logic would actually be in the RowState. See my main comment.